### PR TITLE
add pattern analyzer docs

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
add pattern analyzer docs

### Issues Resolved
Part of https://github.com/opensearch-project/documentation-website/issues/1483 addressed in this PR.

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@udabhas @varun-lodaya Please provide tech review approval to move this PR forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

File: _analyzers/pattern.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+The `pattern` analyzer can be configured using the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`pattern` | Optional | String | A [Java regular expression](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) used to tokenize the input. Default is `\W+`.
+`flags` | Optional | String | A string containing pipe-separated [Java regex flags](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#field.summary) that modify the behavior of the regular expression.
+`lowercase` | Optional | Boolean | Whether to convert tokens to lowercase. Default is `true`.
+`stopwords` | Optional | String or list of strings | A string specifying a predefined list of stopwords (such as `_english_`) or an array specifying a custom list of stopwords. Default is `_none_`.
+`stopwords_path` | Optional | String | The path (absolute or relative to the config directory) to the file containing a list of stop words.
```
Comment (kolchfa-aws) [inline _analyzers/pattern.md:21]:
```suggestion
`stopwords_path` | Optional | String | The path (absolute or relative to the config directory) to the file containing a list of stopwords.
```

File: _analyzers/pattern.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+nav_order: 90
+---
+
+# Pattern analyzer
+
+The `pattern` analyzer allows you to define a custom analyzer that uses a regular expression (regex) to split the input text into tokens. It also provides options for applying regex flags, converting tokens to lowercase, and filtering out stopwords.
+
+## Parameters
+
+The `pattern` analyzer can be configured using the following parameters.
```
Comment (natebower) [inline _analyzers/pattern.md:13]:
```suggestion
The `pattern` analyzer can be configured with the following parameters.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @AntonEliatra Editorial review. Thanks!

**Review by udabhas — APPROVED** [review]
Looks good to me!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**